src/arraystring.rs

```diff
@@ -877,19 +876,14 @@ where
             let ptr = self.array.as_mut_ptr();
             core::ptr::copy(
                 ptr.add(end),
-                ptr.add(str.len().saturating_add(start)),
-                this_len.saturating_sub(end),
+                ptr.add(str.len()).add(start),
+                this_len - end,
             );
             if !str.is_empty() {
                 core::ptr::copy(str.as_ptr(), ptr.add(start), str.len());
             }
         }
-        self.size = self
-            .len()
-            .saturating_add(str.len())
-            .saturating_add(start)
-            .saturating_sub(end)
-            .into_lossy();
+        self.size += str.len() + start + end;
```

Comment:
woah woah, you need to substract the end !

Reply:
Yeah I fixed this on main haha, my bad!